ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezWork.java

```diff
@@ -513,4 +513,22 @@ private void connectUnionWorkWithWork(UnionWork unionWork, BaseWork work, TezWor
     unionWork.addUnionOperators(context.currentUnionOperators);
     context.workWithUnionOperators.add(work);
   }
+
+  /**
+   * If the given reduceWork is the merged work of a MergeJoinWork, return the name of that MergeJoinWork.
+   * Otherwise, return the name of the given reduceWork.
+   */
+  private String getActualOutputWorkName(GenTezProcContext context, ReduceWork reduceWork) {
```

Comment:
could we refactor this with streaming api?
````
  private String getOutputWorkName(GenTezProcContext context, ReduceWork reduceWork) {
    return context.opMergeJoinWorkMap.values().stream()
      .filter(mergeJoinWork -> mergeJoinWork.getBaseWorkList().contains(reduceWork))
      .map(MergeJoinWork::getMainWork)
      .filter(Objects::nonNull)
      .findAny().orElse(reduceWork)
  }    
````

Reply:
Sure, I updated the code.